aws/middleware/private/metrics/metrics.go

```diff
@@ -103,28 +103,29 @@ type MetricContext struct {
 
 // MetricData stores the collected metric data.
 type MetricData struct {
-	RequestStartTime           time.Time
-	RequestEndTime             time.Time
-	APICallDuration            time.Duration
-	SerializeStartTime         time.Time
-	SerializeEndTime           time.Time
-	MarshallingDuration        time.Duration
-	ResolveEndpointStartTime   time.Time
-	ResolveEndpointEndTime     time.Time
-	EndpointResolutionDuration time.Duration
-	InThroughput               float64
-	OutThroughput              float64
-	RetryCount                 int
-	Success                    uint8
-	StatusCode                 int
-	ClientRequestID            string
-	ServiceID                  string
-	OperationName              string
-	PartitionID                string
-	Region                     string
-	RequestContentLength       int64
-	Stream                     StreamMetrics
-	Attempts                   []AttemptMetrics
+	RequestStartTime                 time.Time
+	RequestEndTime                   time.Time
+	APICallDuration                  time.Duration
+	SerializeStartTime               time.Time
+	SerializeEndTime                 time.Time
+	MarshallingDuration              time.Duration
+	ResolveEndpointStartTime         time.Time
+	ResolveEndpointEndTime           time.Time
+	EndpointResolutionDuration       time.Duration
+	GetIdentityStart, GetIdentityEnd time.Time
```

Comment:
I elected to not since I felt it was obvious in the types, but I can change it. Note this API (all of this) is behind a private/"don't use, will break" disclaimer as well.

Reply:
I'd go with consistency rather than make it obvious with the types